versioned_hdf5/hyperspace.py

```diff
@@ -0,0 +1,355 @@
+# Note: this entire module is compiled by cython with wraparound=False
+# See meson.build for details
+
+from __future__ import annotations
+
+import array
+
+import cython
+import numpy as np
+from cython import Py_ssize_t
+
+_empty_tpl = array.array("l", [])
+_py_ssize_t_nbytes = _empty_tpl.itemsize
+# assert that "long" is the same length as void* on the current platform
+# Note: Py_ssize_t is always the same as np.intp
+assert np.intp().nbytes == _py_ssize_t_nbytes
+
+if cython.compiled:  # pragma: nocover
```

Comment:
One of the many things I love about cython pure python mode: you can just disable cythonization and run coverage on it!